src/Microsoft.AspNetCore.Blazor/UIEventHandlerRenderTreeBuilderExtensions.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Blazor.RenderTree;
+
+namespace Microsoft.AspNetCore.Blazor
+{
+    public static class UIEventHandlerRenderTreeBuilderExtensions
```

Comment:
It would be great to include a comment block at the top of this file explaining the scenarios where these overloads are needed. I was finding it hard to figure out without actually commenting them out to see what broke, given that there's also an overload that takes `UIEventHandler` (the base type) directly and has exactly the same logic inside it.

Reply:
In fact I think it would be better still if it could document the overall flow, in terms of how different Razor syntaxes lead to different `AddAttribute` calls, what the different method overload resolution possibilities are, and what this means for the different code paths at runtime. It will be tricky to keep track of all this otherwise, as the implementation is spread out over quite a lot of files.

I know there always a risk that documentation like that tends to rot over time, but on this occasion it's hard to keep track of it at all otherwise. What do you think?

Reply:
Sure